### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by fibini**
Implement the design from the sneak peek wireframes.
- focus on making sure everything functions first.
- Use methods created in the Blog App - processing data in models

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,6 @@
-<p><%= @users %></p>
\ No newline at end of file
+<%@users.each do |user|%>
+<img src = "<%=user.photo%>">
+<h2><%= user.name%></h2>
+<p>number of posts: <%= user.posts_counter%></p>
+<br>
+<%end%>
```
Comment (omar-muhamad) [inline app/views/users/index.html.erb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Implement%20the%20design%20from%20the%20sneak%20peek%20wireframes), Please add basic styling to users index view to make it match the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_root_users.png).

- [x] Clicking on each user suppose to take you to their show page. So please wrap the whole user by `link_to` to do that. See this [article](https://www.rubyguides.com/2019/05/rails-link_to-method/) to understand more.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,3 @@
-<p><%= @specific_user %></p>
\ No newline at end of file
+<%= render "user" %>
+<p><%= @specific_user.bio %></p>
+<p><%= render "post" %></p>
```
Comment (omar-muhamad) [inline app/views/users/show.html.erb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Implement%20the%20design%20from%20the%20sneak%20peek%20wireframes), Please add basic styling to users show view to make it match the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_page.png).

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,10 @@
...
+<%= render "/users/user" %>
+<%@posts.each do |post| %>
+<p>Post# <%=post.id%></p>
+<p><%= post.text %><p>
+Comments: <%= post.comments_counter%>
+Likes: <%= post.likes_counter %>
+<br>
+<%= render partial:"comment", :locals => {:specific_post => post} %>
+<%end%>
+<button>Pagination</button>
```
Comment (omar-muhamad) [inline app/views/posts/index.html.erb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Implement%20the%20design%20from%20the%20sneak%20peek%20wireframes), Please add basic styling to posts index view to make it match the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png).

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1 +1,7 @@
-<p><%= @specific_post %></p>
\ No newline at end of file
+<p>Post# <%= @specific_post.id %> by <%= @user.name %></p>
+<p>Comments: <%= @specific_post.comments_counter%>, Likes: <% @specific_post.likes_counter %></p>
+<p><%= @specific_post.text %><p>
+<%@specific_post.comments.each do |comment|%>
+<%= comment.author.name %>, <%= comment.text %>
+<br>
+<%end%>
```
Comment (omar-muhamad) [inline app/views/posts/show.html.erb]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md#:~:text=Implement%20the%20design%20from%20the%20sneak%20peek%20wireframes), Please add basic styling to Posts show view to make it match the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png).

File: app/views/users/_post.html.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<%@specific_user.recent_post.each do |post|%>
+Post# <%=post.id%>
+<%= post.text%>
+Comments: <%= post.comments_counter%>
+Likes: <%= post.likes_counter%>
+<br>
+<%end %>
+<button>See all posts</button>
```
Comment (omar-muhamad) [inline app/views/users/_post.html.erb]: - [x] Same here, Clicking on `see all posts` button suppose to take you to the user's post index page(view) for this specific user, So please fix that.
[Hint] You can also use `link_to` here.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @fibini,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Gitflow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by aimemalaika — APPROVED** [review]
Hi @fibini, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉

- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

133591850-fdcb7dd3-dd7a-4274-81e7-0e59434148f7

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**